packages/cli/src/WorkflowRunner.ts

```diff
@@ -33,17 +35,28 @@ import * as PCancelable from 'p-cancelable';
 import { join as pathJoin } from 'path';
 import { fork } from 'child_process';
 
+import * as Bull from 'bull';
 
 export class WorkflowRunner {
 	activeExecutions: ActiveExecutions.ActiveExecutions;
 	credentialsOverwrites: ICredentialsOverwrite;
 	push: Push.Push;
+	jobQueue: Bull.Queue;
 
 
 	constructor() {
 		this.push = Push.getInstance();
 		this.activeExecutions = ActiveExecutions.getInstance();
 		this.credentialsOverwrites = CredentialsOverwrites().getAll();
+
+		const executionsMode = config.get('executions.mode') as string;
+
+		if (executionsMode === 'queue') {
+			// Connect to bull-queue
+			const prefix = config.get('queue.bull.prefix') as string;
+			const redisOptions = config.get('queue.bull.redis') as object;
+			this.jobQueue = new Bull('jobs', { prefix, redis: redisOptions });
```

Comment:
What happens if Redis is down or unreachable? We should try to detect it on startup if possible and maybe switch to regular mode if redis is unavailable?

Reply:
hm, think in this case it would be better to exit. After all, is something essential missing and hiding it would be bad as it would then cause problems later as it operates differently and can not deliver the expected performance.

We would have to however test what happens if redis connection gets lost in between and see how we handle that.